cluster/saltbase/salt/kube-master-addons/init.sls

```diff
@@ -37,10 +37,19 @@ master-docker-image-tags:
   file.touch:
     - name: /srv/pillar/docker-images.sls
 
+# Current containervm image by default has both docker and kubelet
+# running. But during cluster creation stage, docker and kubelet
+# could be overwritten completely, or restarted due flag changes.
+# The ordering of salt states for service docker, kubelet and
+# master-addon below is very important to avoid the race between
+# salt restart docker or kubelet and kubelet start master components.
 kube-master-addons:
   service.running:
     - enable: True
     - restart: True
+    - require:
+      - service: docker
+      - service: kubelet
```

Comment:
Please add a brief note here as well so that the explanation stays with the code. Something along the lines of "Don't allow the kubelet that is pre-installed on the container vm to launch any of the side-loaded containers since it is being replaced by salt."

You might also note that once we move these containers to gcr.io with proper release versions, this sideloading will be unnecessary and we will need another way to control this.